keyboards/espectro/keymaps/mac/keymap.c

```diff
@@ -0,0 +1,185 @@
+|          |          |          |                                                        |        |        |   TO   |        |        |        |        |        |        |
+| L GUI    |  L ALT   | L CTL    |                      SPACE                             | R CTL  | R ALT  |LAYER 0 |  LEFT  |  DOWN  | RIGHT  |   0    |   .    |        |
+|__________|__________|__________|________________________________________________________|________|________|________|________|________|________|________|________|________|
+ */
+
+
+	[_WINDOWS] = DEFAULT(
+    KC_ESC,  KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,  KC_F10,  KC_F11,  KC_F12,  KC_DEL,  KC_HOME, KC_END,  KC_PGUP, KC_PGDN, KC_MENU,
+    KC_GRV,  KC_1,   KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,   KC_0,   KC_MINS, KC_EQL,       KC_BSPC,      KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+      KC_TAB,  KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,   KC_LBRC, KC_RBRC,     KC_BSLS,     KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+       KC_CAPS,  KC_A,   KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,   KC_SCLN,   KC_QUOT,          KC_ENT,     KC_P4,   KC_P5,   KC_P6,
+        KC_LSPO,  KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT, KC_SLSH,      KC_RSPC,        KC_UP,   KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+    KC_LGUI, KC_LALT, KC_LCTL,                   KC_SPC,                           KC_RCTL,  KC_RALT,  TO(0),  KC_LEFT, KC_DOWN,  KC_RIGHT, KC_P0,   KC_PDOT
+  ),
+};
+
+
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    // dynamically generate these.
+    case EXPOSE:
+      if (record->event.pressed) {
+        return MACRO(D(LCTRL), T(UP), U(LCTRL), END);
+      }
+      return false;
+      break;
+    case LAUNCH:
+      if (record->event.pressed) {
+        return MACRO(D(LCTRL), T(L), U(LCTRL), END);
+      }
+      return false;
+      break;
+    case DOCK:
+      if (record->event.pressed) {
+        return MACRO(D(LALT), D(LGUI), T(D), U(LALT), U(LGUI), END);
+      }
+      return false;
+      break;
+    case SCRCAP:  //screen capture
+      if (record->event.pressed) {
+        return MACRO(D(LCTRL), T(S), U(LCTRL), END);
+      }
+      return false;
+      break;
+    case CS1:  //custom shortcut 1
+      if (record->event.pressed) {
+        return MACRO(D(LALT), D(LGUI), T(1), U(LALT), U(LGUI), END);
+      }
+      return false;
+      break;
+    case CS2:  //custom shortcut 2
+      if (record->event.pressed) {
+        return MACRO(D(LALT), D(LGUI), T(2), U(LALT), U(LGUI), END);
+      }
+      return false;
+      break;
+    case QALL:  //quit all applications
+      if (record->event.pressed) {
+        return MACRO(D(LALT), D(LGUI), T(Q), U(LALT), U(LGUI), END);
```

Comment:
This should be: 
```suggestion
        SEND_STRING(SS_DOWN(X_LALT) SS_DOWN(X_LGUI) SS_TAP(X_Q) SS_UP(X_LGUI) SS_UP(X_LALT));
```

Reply:
All of these macro changes are causing errors upon compiling:

keyboards/espectro/keymaps/mac/keymap.c:142:1: error: \x used with no following hex digits
         SEND_STRING(SS_DOWN(X_LCTL) SS_TAP(X_UP) SS_UP(X_LCTL));